core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -1977,15 +2034,24 @@ object Stream {
      * res0: List[Int] = List(1, 2, 3, 0)
      * }}}
      */
-    def handleErrorWith[O2>:O](h: Throwable => Stream[F,O2]): Stream[F,O2] =
-      Stream.fromFreeC(self.get[F,O2] handleErrorWith { e => h(e).get })
+    def handleErrorWith[O2>:O](h: Throwable => Stream[F,O2]): Stream[F,O2] = {
+      fromFreeC(Algebra.openScope[F, O2](None) flatMap { scope =>
```

Comment:
This can be rewritten more simply as `Stream.fromFreeC(self.scope.get[F,O2] handleErrorWith { e => h(e).get })`

Reply:
ok

Reply:
after second thougth we need to close the scope also in case of normal termination. But we can write something like Algebra.scope(self).handleErrorWith(...)